### PR TITLE
[AIRFLOW-2086] Add a configuration variable to control numbers of dag run for display

### PR DESCRIPTION
**Pull request by feng-tao**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2086
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Similar to what airflow-2393, airflow-2086 describes, airflow tree view is very slow to run which often triggers guicorn timeout if the dag has many tasks and dependencies. We limit the default tree view to most recent 5 dagruns which help to alleviate the issue.

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by Fokko — CHANGES_REQUESTED** [review]
I would suggest to make this configurable. Right now I only see a lot of magic variables ;)

**Comment by milanvdm** [comment]
I would like to have this included in the upcoming release as we have clients with a poor internet connection and showing 25 dags as default is slowing us down.

Making it configurable is IMHO another kind of PR and is not relevant to this one.

**Comment by feng-tao** [comment]
Hey @Fokko , I update the pr to add a new config for default_dag_run :)

**Comment by feng-tao** [comment]
ping @Fokko and @mistercrunch

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -280,6 +280,9 @@ rbac = False
 # Define the color of navigation bar
 navbar_color = #007A87
 
+# Default dagrun to show in UI
+default_dag_run = 5
```
Comment (artwr) [inline airflow/config_templates/default_airflow.cfg]: `default_dag_run_display_number` maybe?

**Review by artwr — COMMENTED** [review]
@feng-tao I am not completely sure about making the default 5. I agree with a configurable default so that Airflow admins can decide for themselves how many makes sense depending on the typical shape of their DAGs. But a beginner running a small backfill would most likely want a little bit more history than just 5 runs.

**Comment by Fokko** [comment]
I agree with @artwr, maybe set the default to 25?

**Comment by artwr** [comment]
I think the default of 25 is right. If it is really too much we could drop it down to 15 (just over 2 weeks), but then 15 needs to be in the drop-down.

**Comment by feng-tao** [comment]
sure, let me update the pr back to 25 later today.

**Comment by feng-tao** [comment]
hey @artwr and @Fokko , I update the pr by changing the default to 25. Would you mind taking another look? Thanks.

**Comment by artwr** [comment]
LGTM +1

**Pull request closed without merging**